File: class2/BSM-2/01/MainActivity.java
package com.example.loginnotepadjava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText password;
    private Button button;
    public static String correct_password;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        password = findViewById(R.id.password);
        button = (Button) findViewById(R.id.login);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                correct_password = NotepadActivity.new_password;

                //validate password
                if (TextUtils.isEmpty(password.getText().toString())) {
                    Toast.makeText(MainActivity.this, "Empty input", Toast.LENGTH_LONG).show();
                }
                else if (password.getText().toString().equals(correct_password)) {
                    Toast.makeText(MainActivity.this, "Correct password", Toast.LENGTH_LONG).show();
                    openNotepad();
                }
                else
                    Toast.makeText(MainActivity.this, "Invalid password", Toast.LENGTH_LONG).show();
            }
        });
    }
    public void openNotepad() {
        Intent intent = new Intent(this, NotepadActivity.class);
        startActivity(intent);
    }
}

File: class2/BSM-2/06/MainActivity.kt
package com.example.notes

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.submitPassword).setOnClickListener {

            val passwd = findViewById<EditText>(R.id.editTextPassword).text.toString()
            // Toast.makeText(this, passwd, Toast.LENGTH_SHORT).show()
            val dbClient = getSharedPreferences(packageName, Context.MODE_PRIVATE)
            if(dbClient.contains("password")) {
                val storedPasswd = dbClient.getString("password", null)
                // Toast.makeText(this, "Password ${passwd == storedPasswd}", Toast.LENGTH_SHORT).show()
                if(passwd == storedPasswd) {
                    startActivity(Intent(this, Note::class.java))
                } else {
                    Toast.makeText(this, "Password invalid", Toast.LENGTH_SHORT).show()
                }
            } else {
                dbClient.edit().putString("password", passwd).apply()
            }
        }
        findViewById<Button>(R.id.passwdReset).setOnClickListener {
            startActivity(Intent(this, ChangePasswd::class.java))
        }
    }
}
